src/commons/application/types/SessionTypes.ts

```diff
@@ -1,4 +1,5 @@
 import { Grading, GradingOverview } from '../../../features/grading/GradingTypes';
+import { Input } from '../../../features/sourceRecorder/SourceRecorderTypes';
```

Comment:
Move `Input` into `ApplicationTypes.ts`

Reply:
Technically, most of the recording types are defined in `SourceRecorderTypes`, so it wouldn't make much sense to just shift `Input` into `ApplicationTypes` 

If that's the case, we need to shift everything into `ApplicationTypes` which might not be very nice 

Maybe @jiayushe can also advice on this >.<

Reply:
Hmm what I usually do in React projects is to put all the data types in one folder, for example `src/types/<component>.ts`, while leaving the action constants in the individual action files.

Reply:
Gotcha! 

So in this case would it make sense to shift the declared types of `Input` etc etc all to `ApplicationTypes` since we would be using them in `Playground` and `AssessmentWorkspace` now given that we would doing keystroke recording for these 2 components? o.O

Reply:
You can either pull `SourceRecorder` into commons, or just put the `Input` into the `ApplicationTypes`. Point being is `Input` is used by more than one feature, hence should be under `commons/`.

Reply:
👍